lte/gateway/c/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -339,12 +339,15 @@ void clean_stale_enb_state(
   // Remove the S1 context for UEs associated with old eNB association
   hashtable_key_array_t* keys =
       hashtable_uint64_ts_get_keys(&stale_enb_association->ue_id_coll);
-  ue_description_t* ue_ref = NULL;
-  for (int i = 0; i < keys->num_keys; i++) {
-    ue_ref = s1ap_state_get_ue_mmeid((mme_ue_s1ap_id_t) keys->keys[i]);
-    s1ap_remove_ue(state, ue_ref);
+  if (keys != NULL) {
+    ue_description_t* ue_ref = NULL;
+    for (int i = 0; i < keys->num_keys; i++) {
+      ue_ref = s1ap_state_get_ue_mmeid((mme_ue_s1ap_id_t) keys->keys[i]);
+      s1ap_remove_ue(state, ue_ref);
+    }
+    FREE_HASHTABLE_KEY_ARRAY(keys);
   }
-  FREE_HASHTABLE_KEY_ARRAY(keys);
+
   // Remove the old eNB association
   s1ap_remove_enb(state, stale_enb_association);
   update_mme_app_stats_connected_enb_sub();
```

Comment:
Nit comment: Can we move the call to `update_mme_app_stats_connected_enb_sub()` inside `s1ap_remove_enb()` ? 
In this case, the metric will be decremented twice, once when eNB is removed as part of last call to `s1ap_remove_ue()` and once in line 353 here